src/inference/inference.ts

```diff
@@ -34,36 +34,17 @@ export class Inference {
     inputs: Array<string>,
     params: Record<string, string>
   ): Promise<EmbeddingsList> {
-    try {
-      const typedAndFormattedInputs: Array<EmbedRequestInputsInner> =
-        this._formatInputs(inputs);
-      const typedParams: EmbedRequestParameters = this._formatParams(params);
-      const typedRequest: EmbedOperationRequest = {
-        embedRequest: {
-          model: model,
-          inputs: typedAndFormattedInputs,
-          parameters: typedParams,
-        },
-      };
-      const response = await this._inferenceApi.embed(typedRequest);
-      if (response.model && response.data && response.usage) {
-        return new EmbeddingsList(
-          response.model,
-          response.data,
-          response.usage
-        );
-      } else {
-        console.log(
-          'Response from Inference API is missing required fields; response:',
-          response
-        );
-        throw new Error(
-          'Response from Inference API is missing required fields'
-        );
-      }
-    } catch (error) {
-      console.log('Error occurred while generating embeddings:', error);
-      throw error;
-    }
+    const typedAndFormattedInputs: Array<EmbedRequestInputsInner> =
+      this._formatInputs(inputs);
+    const typedParams: EmbedRequestParameters = this._formatParams(params);
+    const typedRequest: EmbedOperationRequest = {
+      embedRequest: {
+        model: model,
+        inputs: typedAndFormattedInputs,
+        parameters: typedParams,
+      },
+    };
+    const response = await this._inferenceApi.embed(typedRequest);
+    return new EmbeddingsList(response.model, response.data, response.usage);
```

Comment:
I don't think we should be throwing or logging errors here. If there's an error from making the request, it should bubble up like our other operations.

See this comment for a bit more context on why I changed this: https://github.com/pinecone-io/pinecone-ts-client/pull/240/files#r1683236813. @aulorbe I wasn't able to directly reproduce what the previous error code was trying to handle, but let me know if there's some way of doing that.